Component versions (#86)
mulesoft/docs-private-cloud: modules/ROOT/pages/config-alerts.adoc

> = Configure Alerts for Anypoint Private Cloud

Reviewer: ```suggestion
= Configure Alerts for Anypoint PCE
```

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

Anypoint Platform Private Cloud Edition (Anypoint Platform PCE) provides built-in alerts that are triggered when a condition specified in any alert definition is detected.

Measurements are stored in https://www.influxdata.com/products/influxdb-overview[Influxdb] and read by https://www.influxdata.com/time-series-platform/kapacitor[Kapacitor]. Kapacitor sends emails when an alert is triggered.

== Alert Definitions

[%header%autowidth.spread]
|===

> | Name | Description | Threshold | Configmap

Reviewer: Is `configmap` syntax? You would not format it in the column heading, of course, unless you call it Config Map or Configuration Map. I wonder if it should be lowercase, however, if it really is syntax.

| High CPU | Measures the total CPU usage on all cluster nodes | Warning 75%, Critical 90% | kapacitor-alerts 
| High Memory | Measures the memory consumption on all cluster nodes | Warning 75%, Critical 90% | kapacitor-alerts 
| Filesystem Usage	| Measures the file system usage | Warning 80%, Critical 90% | kapacitor-alerts 
| Kubernetes Node Status | Checks Kubernetes status for all nodes| When Kubernetes reports a `NotReady` node status | kapacitor-alerts 

> | Etcd Latency | Measures the latency between nodes. Latency can affect `etcd` performance | Warning 521 seconds, Critical 1024 seconds | kapacitor-alerts 

Reviewer: ```suggestion
| Etcd Latency | Measures the latency between nodes, which can affect `etcd` performance | Warning 521 seconds, Critical 1024 seconds | kapacitor-alerts 
```

| Systemd health | Checks if the `systemd` slice (where gravity is running) is healthy | When gravity or child systems, such as `docker`, `etcd`, `kube-proxy`, `kube-kubelet`, `planet-agent`, `dnsmasq`, and `flanneld`, are down | kapacitor-alerts 
| Influxdb Healthcheck Status | Checks the health of `Influxdb` (where the metrics for monitoring are stored) | No connection to `Influxdb` | kapacitor-alerts 
| Networking Configuration | Checks networking operating system flag status | When `ip_forward` or `br_netfilter` flags are disabled | kapacitor-alerts 

> | Node Uptime | Checks if nodes are online | When a node is down for more than 5 minutes | kapacitor-alerts 

Reviewer: picky....
```suggestion
| Node Uptime | Checks if nodes are online | When a node is down for more than five minutes | kapacitor-alerts 
```

| Etcd Health | Check the status of `etcd` | When `etcd` is down or not healthy | kapacitor-alerts 

> | Docker Status	| Verifies the docker service is running on each node | When docker service is not running on a node | kapacitor-alerts 

Reviewer: ```suggestion
| Docker Status	| Verifies that the docker service is running on each node | When docker service is not running on a node | kapacitor-alerts 
```

> | Backup Failing | Check the result of a backup operation | When a backup operation fails | backup-status-alert

Reviewer: ```suggestion
| Backup Failing | Checks the result of a backup operation | When a backup operation fails | backup-status-alert
```

> | Stolon Replication Lag | Check `Postgres` replicas are in synch | When a replica is lagging by 1800 or more seconds | kapacitor-alerts 

Reviewer: ```suggestion
| Stolon Replication Lag | Verifies that `Postgres` replicas are in sync | When a replica is lagging by 1800 or more seconds | kapacitor-alerts 
```

> | Cassandra Load | Verifies Cassandra nodes are in sync | Warning 10% difference, Critical 20% difference | cassandra-load

Reviewer: ```suggestion
| Cassandra Load | Verifies that Cassandra nodes are in sync | Warning 10% difference, Critical 20% difference | cassandra-load
```

|===

== Configure Alert Definitions 

. From OpsCenter, navigate to *Configuration*.
. Select the *monitoring* namespace.

> . Select a configmap and apply your changes.

Reviewer: configmap looks like syntax: config map? configuration map? or...
```suggestion
. Select a `configmap` instance and apply your changes.
```
OR
```suggestion
. Select `configmap` and apply your changes.
```

> If you apply changes to the `kapacitor-alerts` configmap, you must restart Kapacitor. Follow the instructions in the Restart Kapacitor section in this topic.

Reviewer: ditto re. configmap

Reviewer: Avoid referring to topics and sections (self-reference; see style reference):
```suggestion
If you apply changes to the `kapacitor-alerts` configmap, you must restart Kapacitor. Follow the instructions in the Restart Kapacitor text.
```

== Configure Alerts

. Configure the `FROM` and `TO` email addresses:
.. Log in to the Ops Center console and then select *Configuration*.

> .. From the *Namespace* drop-down, select *monitoring*.

Reviewer: ```suggestion
.. From the *Namespace* drop-down list, select *monitoring*.
```
OR
```suggestion
.. From the *Namespace* list, select *monitoring*.
```

> .. From the *Config maps* drop-down, select *alerting-addresses*.

Reviewer: ditto re. "drop-down," GLOBALLY. I will stop marking in this PR.

.. Select *From* and then enter the `FROM` email address used for the alert.
.. Select *To* and then enter the `TO` email address used for the alert.

> . Configure your SMTP email server to send alerts. Configuring the SMTP server in Anypoint Access Management restarts the Anypoint Platform service responsible for initiating alerts.

Reviewer: ditto list item length:
```suggestion
. Configure your SMTP email server to send alerts. 
+
+ Configuring the SMTP server in Anypoint Access Management restarts the Anypoint Platform service responsible for initiating alerts.
```

. Verify alerts configuration:
+

> ** Verify that your SMTP server can send and receive emails using the addresses you configure as the `FROM` and `TO` addresses in the following section.

Reviewer: ditto self-reference:
```suggestion
** Verify that your SMTP server can send and receive emails using the addresses you configure as the `FROM` and `TO` addresses.
```

> ** Verify that your cluster nodes can communicate with your SMTP server. For example, use `telnet` to connect to your SMTP server from one of your cluster nodes:

Reviewer: ```suggestion
** Verify that your cluster nodes can communicate with your SMTP server. 
+
+ For example, use `telnet` to connect to your SMTP server from one of your cluster nodes:
```

+
----
telnet my.smtp.server.com 587
Trying XXX.XXX.XXX.XXX...
Connected to my.smtp.server.com.
Escape character is '^]'.
220 my.smtp.server.com ESMTP
^[^]
telnet> quit
Connection closed.
----

> 

Reviewer: BREAK HERE to prevent GitHub issue
